### PR TITLE
[FIX] hr_org_chart: employee form view flickers on edit and discard

### PR DESCRIPTION
**Pull request by dhp-odoo**
PURPOSE:

On a large screen (with the chatter displayed on the right of form views),
Go to Hr employee form view >> switch to edit mode  >>  view flickers
As like the chatter disappears for a while and display again. The same issue
occurs when you discard the changes.

SPEC:

Flickers occur as the widget for hr_org_chart is returning the promise and once the
promise resolve then chatter will display. so we have now removed the return of the 
promise and the org_chart will display even after the chatter load and once the render
method get the required data.

Task : 2243856

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/js/form_renderer.js [hunk 1]
```diff
@@ -44,25 +44,27 @@ FormRenderer.include({
...
             if (this.mode === 'edit' && !this.state.data.id) {
+                if (this.chatter) {
+                    // Detach the chatter before updating the $el.
+                    // This is important because if the view is now in create mode
+                    // (edit mode with no res_id), the chatter will be removed from
+                    // the DOM, and its handlers will be unbound. By detaching it
+                    // beforehand, we ensure to keep its handlers alive so that if
+                    // it is re-appended later, everything will still work properly
+                    this.chatter.$el.detach();
+                }
```
Comment (aab-odoo) [inline addons/mail/static/src/js/form_renderer.js:67]: you should never detach the here, as this will cause a flickering

**Review by aab-odoo — DISMISSED** [review]
Wasn't my fix enough? I see you did other changes. Also, could you write a test? You'll need an async widget in the form view. You'll have to check that the chatter is always in the DOM, even during an async re-rendering.

File: addons/mail/static/src/js/form_renderer.js (hunk 1 above)
Comment (msh-odoo) [reply addons/mail/static/src/js/form_renderer.js]: @aab-odoo Flickering does not occur in Create case, I tried but I did not face flickering issue, but still I improved code as we want to detach chatter element in create/edit mode and we will replace temp hook with chatter element only if record has ID, as in create mode we will not have temp hook.

**Comment by aab-odoo** [comment]
@dhp-odoo @msh-odoo could you re-target to saas-13.5? The issue isn't blocking, so I wouldn't fix this in a stable version. Having this fixed for v14 is enough I think.

**Comment by Polymorphe57** [comment]
Hello @dhp-odoo @msh-odoo, could you re-target the PR to master (see aab comment above)?

**Comment by Polymorphe57** [comment]
Task Cancelled --> We close this PR
